manifest/provider/validate.go

```diff
@@ -102,6 +103,42 @@ func (s *RawProviderServer) ValidateResourceTypeConfig(ctx context.Context, req
 		}
 	}
 
+	// check for lists with inconsistent types
+	tftypes.Walk(manifest, func(ap *tftypes.AttributePath, v tftypes.Value) (bool, error) {
+		var elems []tftypes.Value
+		if err := v.As(&elems); err != nil {
+			return true, nil
+		}
+		first := elems[0]
+		for _, e := range elems[1:] {
+			// FIXME I had to do this so I could actually tell the user
```

Comment:
@alexsomesan I could use some help with this because I was scratching my head trying to figure out how to actually direct the user to a field in the manifest in the diagnostic output. Nothing I tried to seems to work.